include/boost/json/parser.hpp

```diff
@@ -104,6 +104,21 @@ class parser : public basic_parser
     BOOST_JSON_DECL
     parser();
 
+    /** Constructor.
+
+        Constructs a parser using the supplied
+        @ref storage_ptr to allocate internal storage.
+
+        @note
+        Before any JSON can be parsed, the function
+        @ref start must be called.
+
+        @param sp The @ref storage_ptr to use for
+        allocating internal storage.
+    */
+    BOOST_JSON_DECL
+    explicit parser(storage_ptr sp);
```

Comment:
explicit on its own line please

Reply:
We should call it "intermediate storage" and make it clear that this is used internally and not for the final DOM. We also need to explain that it is reused between parses, which improves performance.

Reply:
Will do